Sources/SwiftSyntaxMacros/MacroProtocols/MemberMacro.swift

```diff
@@ -28,4 +28,39 @@ public protocol MemberMacro: AttachedMacro {
     providingMembersOf declaration: some DeclGroupSyntax,
     in context: some MacroExpansionContext
   ) throws -> [DeclSyntax]
+
+  /// Expand an attached declaration macro to produce a set of members.
+  ///
+  /// - Parameters:
+  ///   - node: The custom attribute describing the attached macro.
+  ///   - declaration: The declaration the macro attribute is attached to.
+  ///   - missingConformancesTo: The set of protocols that were declared
+  ///     in the set of conformances for the macro and to which the declaration
+  ///     does not explicitly conform. The member macro itself cannot declare
+  ///     conformances to these protocols (only an extension macro can do that),
+  ///     but can provide supporting declarations, such as a required
+  ///     initializer or stored property, that cannot be written in an
+  ///     extension.
+  ///   - context: The context in which to perform the macro expansion.
+  ///
+  /// - Returns: the set of member declarations introduced by this macro, which
+  /// are nested inside the `attachedTo` declaration.
+  static func expansion(
+    of node: AttributeSyntax,
+    providingMembersOf declaration: some DeclGroupSyntax,
+    missingConformancesTo protocols: [TypeSyntax],
+    in context: some MacroExpansionContext
+  ) throws -> [DeclSyntax]
+}
+
+public extension MemberMacro {
+  /// Default implementation that ignores the unhandled conformances.
+  static func expansion(
+    of node: AttributeSyntax,
+    providingMembersOf declaration: some DeclGroupSyntax,
+    missingConformancesTo protocols: [TypeSyntax],
+    in context: some MacroExpansionContext
+  ) throws -> [DeclSyntax] {
+    return try expansion(of: node, providingMembersOf: declaration, in: context)
+  }
 }
```

Comment:
AFAICT this means that you always need to implement `expansion(of:providingMembersOf:in:)`, even if you implement `expansion(of:providingMembersOf:missingConformancesTo:in:)`. I don’t have a better idea but just want to put this fact out there.

Reply:
Yeah, I can create a default implementation in the other direction and deprecate one.